src/server/game/Entities/Object/Object.cpp

```diff
@@ -2567,7 +2550,26 @@ float WorldObject::GetFloorZ() const
 {
     if (!IsInWorld())
         return m_staticFloorZ;
-    return std::max<float>(m_staticFloorZ, GetMap()->GetGameObjectFloor(GetPhaseMask(), GetPositionX(), GetPositionY(), GetPositionZ()));
+    return std::max<float>(m_staticFloorZ, GetMap()->GetGameObjectFloor(GetPhaseMask(), GetPositionX(), GetPositionY(), GetPositionZ() + GetCollisionHeight()));
+}
+
+float WorldObject::GetMapWaterOrGroundLevel(float x, float y, float z, float* ground/* = nullptr*/) const
+{
+    if (Unit const* unit = ToUnit())
+    {
+        *ground = z;
```

Comment:
is this some new code that doesn't handle the case of ground = nullptr or was it just copied from somewhere ? I'd prefer to handle that case too to avoid crashes (or ASSERT() is ground is never supposed to be null)

Reply:
It should be removed and let Map::GetWaterOrGroundLevel deal with it